native/src/main/java/io/ballerina/stdlib/sql/utils/Utils.java

```diff
@@ -317,6 +318,7 @@ private static Type validFieldConstraint(int sqlType, Type type) {
 
     public static PrimitiveTypeColumnDefinition getColumnDefinition(ResultSet resultSet, int columnIndex, Type type)
             throws SQLException, ApplicationError {
+        type = TypeUtils.getReferredType(type);
```

Comment:
Lets add this check to nativeQueryRowExecutable() as discussed. I see union check is done before this code execution, which will fail if it is a describing type. This should be done in other native code levels as well.

Reply:
Fixed in [114414c](https://github.com/ballerina-platform/module-ballerina-sql/pull/543/commits/114414c1d01ded9c591b1498dbae9c917851c032)

Reply:
Missed in, 

https://github.com/kaneeldias/module-ballerina-sql/blob/114414c1d01ded9c591b1498dbae9c917851c032/native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java#L124

https://github.com/kaneeldias/module-ballerina-sql/blob/114414c1d01ded9c591b1498dbae9c917851c032/native/src/main/java/io/ballerina/stdlib/sql/utils/Utils.java#L344


The above are only for query() function, check the same instances in call procedure as well

Reply:
We have to use referredType() even for the record descriptions for the steam. As it self can refer to another record.

Reply:
Fixed in [1331c45](https://github.com/ballerina-platform/module-ballerina-sql/pull/543/commits/1331c45b51d3f1cf3ae703b3147e2afce8be9d24)